File: 2396-strictly-palindromic-number/2396-strictly-palindromic-number.py
class Solution:
    def palindrome(self, n):
        s = bin(n)[2:]  
        start = 0
        end = len(s) - 1
        
        while start < end:
            if s[start] != s[end]:
                return False
            start += 1
            end -= 1
        
        return True

    def isStrictlyPalindromic(self, n: int) -> bool:
        num = n
        
        for i in range(2, n-1):
            if num == 0:
                return False
            if not self.palindrome(num):
                return False
            num //= 2
        
        return True

File: 2396-strictly-palindromic-number/2396-strictly-palindromic-number.java
class Solution {
    public boolean palindrome(int n){
        String str=Integer.toBinaryString(n);
        int s=0, e=str.length()-1;
        while(s<e){
            if(str.charAt(s)!=str.charAt(e))return false;
            s++;
            e--;
        }
        return true;
    }
    public boolean isStrictlyPalindromic(int n) {
        int num=n;
        for(int i=2; i<n-1; i++){
            if(num==0)return false;
            if(!palindrome(num))return false;
            num/=2;
        }
        return true;
    }
}